packages/editor/src/components/post-visibility/check.js

```diff
@@ -23,11 +18,7 @@ export default compose( [
 	withSelect( ( select ) => {
 		const { getCurrentPost, getCurrentPostType } = select( editorStore );
 		return {
-			hasPublishAction: get(
-				getCurrentPost(),
-				[ '_links', 'wp:action-publish' ],
-				false
-			),
+			hasPublishAction: getCurrentPost()._links?.[ 'wp:action-publish' ],
```

Comment:
`getCurrentPost()` will always be an object:

https://github.com/WordPress/gutenberg/blob/4944315cac156c78313121657dff9d58ce8f4d3f/packages/editor/src/store/selectors.js#L170